### ISSUE TITLE
External screens not available (until restart of Flameshot)

### ISSUE THREAD

**Issue by tpcgold** [flameshot-org/flameshot#3505]
### Flameshot Version

v12.1.0 (96c2c82e)

### Installation Type

`brew install --cask flameshot`

### Operating System type and version

Soonoma 14.3.1

### Description

Flameshot area doesn't extend to external screens when plugging them in. 
So taking a screenshot or "selection" only involves the main display of the M2 Mac. 

Tried to reinstalled etc. but the problem still occures (this problem exists since a few months)

Current workaround found:  close flameshot and reopen -> but this is really annoying over time

### Steps to reproduce

1. start Mac
2. add external Screen via usb-c/thunderbolt
3. problem occurs

### System Information

1. macOS Soonoma 14.3.1
2. Mac M2 with two Dell external Screens or Mac M2 with Samsung external Screen

**Comment by lateparty** [comment]
Second this on Windows. Flameshot snapshots your multi-screen configuration on startup and then keeps that regardless of plugging or unplugging and moving to different setups with the device. Would love a fix for this.

**Comment by tpcgold** [comment]
seems the problem could be fixed by using:
https://doc.qt.io/qt-6/qguiapplication.html#screenAdded

(but I'm not a contributor atm. and don't have time to read into the code. just this trigger doesn't occur in the code anywhere , although it seems it should)
Also this trigger should occur:
https://doc.qt.io/qt-6/qguiapplication.html#screenRemoved

**Comment by man0s** [comment]
+1

**Comment by johannesE** [comment]
I can confirm that restarting flameshot on OSX fixed the issue for me.

**Comment by Saibamen** [comment]
>fixed the issue for me

This is a workaround, not fixing

**Comment by leehambley** [comment]
I'm trying today to apply the advice from https://github.com/flameshot-org/flameshot/issues/3505#issuecomment-1965879269 and see if that works in general, if it does I'll link my pull request here. First impressions are good, but I'm not fluent in C++ or QT so I can't promise anything.

**Comment by leehambley** [comment]
PR [here](https://github.com/flameshot-org/flameshot/pull/3618) but I don't know if it really really helps. You can try cloning my branch and building with 

```
brew install qt@5 cmake
cmake -S . -B build -DQt5_DIR=$(brew --prefix qt5)/lib/cmake/Qt5 && (cd build && make)
open build/src/flameshot.app
```

**Comment by RichDCs** [comment]
@leehambley Tried it, it does detect the external monitor (even after unplugging and plugging again) 👍

However, it introduces a couple of regressions:
- The screenshot capture on the external monitor does not show any open windows. You can only screenshot an empty screen.
- It adds an additional window on the main monitor - to now dismiss the screenshot mode, you have to press esc -> click on the main monitor to focus -> press esc again.

**Comment by nikhilweee** [comment]
Came here to say that I love Flameshot but this has been bugging me for a while now. The most annoying part is when I right click on the taskbar icon to quit flameshot but I can only see the first menu item because the rest of it is off the screen. Would love to see some progress here!

**Comment by tpcgold** [comment]
seems the development of flameshot is death... so I'll have to look for alternatives 🤷

**Comment by kave** [comment]
> seems the development of flameshot is death... so I'll have to look for alternatives 🤷

let me know if you find a good one!

**Comment by mmahmoudian** [comment]
> seems the development of flameshot is death... so I'll have to look for alternatives 🤷 

Yes, the development have slowed down. Partly since our main dev stepped down, and partly because we received death threats (yes, genuinely) and nasty messages from users. People don't [want to] understand that we are all volunteers in this project, we don't accept donations nor we we get money working on this.

We have asked users to put bounty on the issues they need to solve, this way their money directly translates to actual tangible improvement, but it also attracts people to contribute to the project.

https://flameshot.org/donate/

So far the nasty comments are still coming in, and not much bounty have been put on issues. So we develop the project with the pace that we can without getting burn-out. I hope this make sense.

**Comment by Saibamen** [comment]
Issues are one thing, but the other is this: no new releases:

![Image](https://github.com/user-attachments/assets/e8ca2c1f-e794-4a1d-8af3-8bc1accad0b5)

**Comment by tpcgold** [comment]
@borgmanJeremy more like
https://github.com/flameshot-org/flameshot/issues/3597
is a duplicate as created on May 7, 2024
